content/blog/learnings-from-clean-coder.md

```diff
@@ -0,0 +1,70 @@
++++
+title = "Learnings From Clean Coder"
+slug = "demo-post"
+date = 2022-04-29T11:55:43+05:30
+image = "/images/2021/07/demo.jpg"
+draft = true
+author = "Incubyte"
+authorLink = "https://incubyte.co"
+description = ""
+tags = ["Playbook", "Software Craftsmanship"]
+categories = ["Playbook", "Software Craftsmanship"]
+type = ""
++++
+
+## Learnings from Robert “Uncle Bob” Martin
+
+Robert Martin, fondly known as “Uncle Bob” in the software community, is a software engineer and more famously an author and the founding fathers of many software movements like software craftsmanship, agile methodology and test-driven development. Some of his most famous works include Clean Code, The Clean Coder, Clean Architecture, and Clean Agile. Martin is also a frequent instructor, with the ability to boil down difficult topics into simple, understandable format, along with an occasional cheeky humour.
+One of the most striking qualities of Martin – in all of his writing and lectures, is that nowhere he claims to have a full-proof solution or a doctrine which could solve all the software development problems. He acknowledges that the teachings and lessons are the ones he acquired from his 40+ years of experience, and encourages the reader to validate the “time-testedness” and accuracy of those lessons. One of the qualities of a software craftsman, according to Martin, is to adapt to the changing times and let go of the principles which are irrelevant.
+This article can be thought of as a list of learnings from the book Clean Code. By no means this is an exhaustive list, but rather a list of all the lessons which had the most impact while reading the book.
+
+### 1. The Meaning of Clean Code
+
+Uncle Bob starts off the book Clean Code by exploring the term itself. Without stating a single clear-cut definition for the term, the author presents varied definitions from the perspective of different computer scientists. Reading through them, one can figure out common themes and characteristics appear again and again – readability, straightforward, maintainable, reads like a prose, and doesn’t contain duplication. Thus, we infer that clean code has more to do with writing understandable code which is both easily maintainable and changeable, against clever logic which is harder to maintain.
+
+### 2. Duplication is the Evil
+
+Uncle Bob goes on a length about various characteristics of bad code. Amongst them, code duplication holds a special spot, because majority of code problems have their origins linked to duplication. It has inspired many coding patterns, techniques and ideologies. For e.g., the database normalization techniques (1NF, 2NF, 3NF…) by Codd is ultimately about removing redundancy in the database design. In fact, paradigms like OOP, AOP, COP, Structured Programming, are all different ways to deal with duplication. So, one of the primary tasks while writing a piece of software is to be mindful about code duplication.
```

Comment:
Isn't it also true that some level of duplication may be fine depending on the reason and context? If yes, maybe you want to add it here?